src/box/ck_constraint.h

```diff
@@ -76,6 +76,10 @@ struct ck_constraint_def {
 	 * disabled (set to false) contraint won't be fired.
 	 */
 	bool is_enabled;
+	/** True if this is a field constraint, false otherwise. */
+	bool is_field_ck;
+	/** Fieldno of the field contains the constraint. */
+	uint32_t fieldno;
```

Comment:
Sorry, I just noticed it now. These fields you apparently added to store here some temporary data during parsing. That is a problem because the def is also stored in `ck_constraint` itself and these fields are left uninitialized there. I.e. if I would fetch these cks from `space->format`, the flag `is_field_ck` will be false even for field checks.

The suggestions are:
- In SQL parser introduce your own `sql_ck_constraint_def` of some sort with any additional fields. Or reuse `ck_constraint_parse`;
- Store the field and tuple checks in separate rlists. Don't know how to resolve the fieldno though. I am not sure you really need it. Store it all in `create_column_def` somehow? Created columns are ordered. Which means you must be able to learn ck fieldno by its parent column fieldno.

Reply:
I do not think that this is a problem. I plan to drop all code of old FK and CK as a follow-up PR, where I will drop or move all these structures. I think that there is no need to rework them in this PR.

Reply:
If you say so.